app/models/mixins/miq_web_server_worker_mixin.rb

```diff
@@ -13,48 +13,26 @@ def self.binding_address
       BINDING_ADDRESS
     end
 
-    def self.rails_server
-      VMDB::Config.new("vmdb").config.fetch_path(:server, :rails_server) || "thin"
-    end
+    def self.preload_for_worker_role
+      require 'hamlit-rails'
 
-    def self.rails_server_command_line
-      @rails_server_command_line ||= begin
-        "#{nice_prefix} #{Rails.root.join("bin", "rails")} server #{rails_server}".freeze
-      end
+      # Make these constants globally available
+      ::UiConstants
+
+      configure_secret_token
     end
 
-    def self.build_command_line(*params)
-      params = params.first || {}
+    def self.configure_secret_token
+      Vmdb::Application.config.secret_token = MiqDatabase.first.session_secret_token
 
-      defaults = {
-        :port        => 3000,
-        :binding     => binding_address,
-        :environment => Rails.env.to_s,
-        :config      => Rails.root.join("config.ru")
-      }
+      # To set a secret token after the Vmdb::Application is initialized?,
+      # we need to reset the secrets since they are cached:
+      # https://github.com/rails/rails/blob/4-2-stable/railties/lib/rails/application.rb#L386-L401
+      Vmdb::Application.secrets = nil if Vmdb::Application.initialized?
```

Comment:
We need to reset the secrets here because forked processes for the UI/WS worker needs to configure the secret_token after the main process has already initialized the Rails::Application, the secrets are cached, and we just provided a new secret_token.